util/dvsim/LauncherFactory.py

```diff
@@ -43,24 +41,24 @@ def set_launcher_type(is_local=False):
     elif launcher == "lsf":
         launcher_cls = LsfLauncher
 
-        # The max_parallel setting is not relevant when dispatching with LSF.
-        Scheduler.max_parallel = sys.maxsize
-
     # These custom launchers are site specific. They may not be committed to
     # the open source repo.
     elif launcher == "edacloud" and EDACLOUD_LAUNCHER_EXISTS:
         launcher_cls = EdaCloudLauncher
 
-        # The max_parallel setting is not relevant when dispatching with
-        # EDACloud.
-        Scheduler.max_parallel = sys.maxsize
-
     else:
         log.error("Launcher {} set using DVSIM_LAUNCHER env var does not "
                   "exist. Using local launcher instead.".format(launcher))
         launcher_cls = LocalLauncher
 
 
+def get_launcher_cls():
+    '''Returns the chosen launcher class.'''
+    global launcher_cls
+    assert launcher_cls is not None
+    return launcher_cls
```

Comment:
You shouldn't need the `global` here, I don't think. Python's scoping rules are weird, but if a read happens before the first write (IIRC), we'll get the global without needing a hint.

However, could you rename `launcher_cls` to `LAUNCHER_CLS`? The usual Python convention is that module-level globals are capitalized. Also, since this is supposed to be accessed through the getter, maybe it should be `_LAUNCHER_CLS` (to make it clear this is an internal variable, not meant to be exported by the module).

Reply:
Done.